pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java

```diff
@@ -434,12 +605,29 @@ public NamespaceIsolationData getNamespaceIsolationPolicy(@PathParam("cluster")
 
     @GET
     @Path("/{cluster}/namespaceIsolationPolicies/brokers/{broker}")
-    @ApiOperation(value = "Get a broker with namespace-isolation policies attached to it", response = BrokerNamespaceIsolationData.class)
-    @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission"),
-            @ApiResponse(code = 404, message = "Namespace-isolation policies/ Broker not found"),
-            @ApiResponse(code = 412, message = "Cluster doesn't exist") })
-    public BrokerNamespaceIsolationData getBrokerWithNamespaceIsolationPolicy(@PathParam("cluster") String cluster,
-            @PathParam("broker") String broker) {
+    @ApiOperation(
+        value = "Get a broker with namespace-isolation policies attached to it",
+        response = BrokerNamespaceIsolationData.class,
+        notes = "This operation requires Pulsar super-user privileges."
+    )
+    @ApiResponses(value = {
+        @ApiResponse(code = 403, message = "Don't have admin permission"),
+        @ApiResponse(code = 404, message = "Namespace-isolation policies/ Broker not found"),
```

Comment:
Do you mean the error code "404" apply to both of the error messages "Namespace-isolation policies/ Broker not found"?

Reply:
yes in either case, pulsar returns 404